### ISSUE TITLE
stack install: MissingH dependency issue

### ISSUE THREAD

**Issue by mgalgs** [aviaviavi/toodles#22]
I've never used `stack` or Haskell for that matter, so maybe I did something obviously wrong.

When I ran `stack install toodles` I got the following error:

```
$ stack install toodles
Writing implicit global project config file to: /home/mgalgs/.stack/global-project/stack.yaml
Note: You can change the snapshot via the resolver field there.
Using latest snapshot resolver: lts-12.14
Downloaded lts-12.14 build plan.
Preparing to install GHC (tinfo6) to an isolated location.
This will not interfere with any system-level installation.
Downloaded ghc-tinfo6-8.4.3.
Installed GHC.
Selected mirror https://s3.amazonaws.com/hackage.fpcomplete.com/
Downloading root
Selected mirror https://s3.amazonaws.com/hackage.fpcomplete.com/
Downloading timestamp
Downloading snapshot
Downloading mirrors
Cannot update index (no local copy)
Downloading index
Updated package index downloaded
Update complete
Populated index cache.

Error: While constructing the build plan, the following exceptions were encountered:

In the dependencies for toodles-0.1.0.11:
    MissingH-1.4.1.0 from stack configuration does not match ==1.4.0.1  (latest matching version is 1.4.0.1)
needed since toodles is a build target.

Some different approaches to resolving this:

  * Set 'allow-newer: true' in /home/mgalgs/.stack/config.yaml to ignore all version constraints and build anyway.

  * Consider trying 'stack solver', which uses the cabal-install solver to attempt to find some working build configuration. This can be convenient when dealing with many complicated constraint
    errors, but results may be unpredictable.

  * Recommended action: try adding the following to your extra-deps in /home/mgalgs/.stack/global-project/stack.yaml:

MissingH-1.4.0.1@sha256:0925ca46d3cad0dc6d35bdfb28c9bbd065ea7c8fd49aea8c572e846f49605e13

Plan construction failed.
```

I took the "Recommended action" and added that guy to my `~/.stack/global-project/stack.yaml`:

```
packages: []
resolver: lts-12.14
extra-deps:
  - MissingH-1.4.0.1@sha256:0925ca46d3cad0dc6d35bdfb28c9bbd065ea7c8fd49aea8c572e846f49605e13
```

After doing that I was able to run `stack install toodles` again and it built successfully and I was able to run `toodles` just fine.

## System info

I had just freshly installed `stack` via [their recommended instructions](https://docs.haskellstack.org/en/stable/README/#how-to-install).  Full steps:

```
$ curl -sSL https://get.haskellstack.org/ | sh
$ stack install toodles
```

This is a Fedora 27 box.

**Comment by aviaviavi** [comment]
Thanks for the detailed report! I got one other report of this for someone else as well. Will look into it shortly

**Comment by aviaviavi** [comment]
The fix is checked in, will publish the new package version soon

**Comment by aviaviavi** [comment]
@mgalgs the fix has been released (version 0.1.0.12), please re-open if you still have issues when trying to install the new version. (you may need to do a `stack update`).

Thanks again for bringing this up!
